packages/belos/src/BelosSolverFactory.hpp

```diff
@@ -783,20 +569,20 @@ isSupported (const std::string& solverName) const
   std::pair<std::string, bool> aliasResult =
     Details::getCanonicalNameFromAlias (solverNameUC);
   const std::string candidateCanonicalName = aliasResult.first;
-  const bool isAnAlias = aliasResult.second;
-
-  // Get the canonical name.
-  const Details::EBelosSolverType solverEnum =
-    Details::getEnumFromCanonicalName (isAnAlias ?
-                                       candidateCanonicalName :
-                                       solverNameUC);
   const bool validCanonicalName =
-    (solverEnum != Details::SOLVER_TYPE_UPPER_BOUND);
+    (get_solverManagers().find(candidateCanonicalName) != get_solverManagers().end());
   return validCanonicalName;
 }
 
 } // namespace Impl
 } // namespace Belos
 
+// We have things like BelosSolverFactory_Tpetra.hpp which are automatically
+// included through the adapter includes to maintain backwards compatibility.
+// The Belos version itself doesn't have a place like that so included here
+// which is awkward. It might make more sense to just copy that code here but
+// it has symmetry with the other files and wanted to preserveve that. To discuss.
```

Comment:
I'm OK with this; the symmetry is helpful.  Also, "preserve" not" "preserveve" :-)

Reply:
Along with above fix, repaired this typo and also 2 warnings due to #endif errors.